components/provisioner/assets/database/provisioner.sql

```diff
@@ -0,0 +1,91 @@
+-- Cluster
+
+CREATE TABLE cluster
+(
+    ID uuid PRIMARY KEY CHECK (id <> '00000000-0000-0000-0000-000000000000'),
+    kubeconfig text,
+    terraform_state json,
+    creation_timestamp timestamp without time zone NOT NULL
+);
+
+
+-- Cluster Config
+
+CREATE TYPE infrastructure_provider AS ENUM (
+    'GARDENER',
+    'GCP',
+    'AKS'
+    );
+
+CREATE TABLE cluster_config
```

Comment:
Just wondering, maybe splitting the cluster options into separate fields is not the best option. I would suggest a slightly different approach. You can persist the infrastructure providers together with some schema of their respective options. Then, when storing the cluster config selecting the provider you will have the appropriate schema to validate the options. This approach will make it more resistant to further changes.